frontend/catalyst/jax_primitives.py

```diff
@@ -549,76 +552,188 @@ def _apply_registered_pass_lowering(
 
 
 #
-# func
+# module
 #
-
-
-@func_p.def_impl
-def _func_def_impl(ctx, *args, call_jaxpr, fn, call=True):  # pragma: no cover
-    raise NotImplementedError()
-
-
-def _func_def_lowering(ctx, fn, call_jaxpr, name_stack) -> str:
-    """Create a func::FuncOp from JAXPR."""
+def lower_callable_to_funcop(ctx, _callable, call_jaxpr):
```

Comment:
I use `_callable` as opposed to `callable` because I've had the experience that [`callable` is itself available in python builtin.](https://docs.python.org/3/library/functions.html#callable) This is only true in some versions of python. This will then I think trigger the linter. I guess I can use func, but it just doesn't sound as nice :sweat_smile:

Reply:
I guess `callable_` would be the equivalent that avoids the convention, or `callable_obj` or something 🤔

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1144/commits/15aa35936a953d1a1836fd9db5aa6334fda31415